packages/remix-dev/config/defaults/entry.server.node.tsx

```diff
@@ -69,71 +31,41 @@ function handleBotRequest(
         abortDelay={ABORT_DELAY}
       />,
       {
-        onAllReady() {
-          shellRendered = true;
-          const body = new PassThrough();
-          const stream = createReadableStreamFromReadable(body);
+        onShellReady() {
+          if (!prohibitOutOfOrderStreaming) {
+            shellRendered = true;
+            const body = new PassThrough();
+            const stream = createReadableStreamFromReadable(body);
 
-          responseHeaders.set("Content-Type", "text/html");
+            responseHeaders.set("Content-Type", "text/html");
 
-          resolve(
-            new Response(stream, {
-              headers: responseHeaders,
-              status: responseStatusCode,
-            })
-          );
+            resolve(
+              new Response(stream, {
+                headers: responseHeaders,
+                status: responseStatusCode,
+              })
+            );
 
-          pipe(body);
-        },
-        onShellError(error: unknown) {
-          reject(error);
-        },
-        onError(error: unknown) {
-          responseStatusCode = 500;
-          // Log streaming rendering errors from inside the shell.  Don't log
-          // errors encountered during initial shell rendering since they'll
-          // reject and get logged in handleDocumentRequest.
-          if (shellRendered) {
-            console.error(error);
+            pipe(body);
           }
         },
-      }
-    );
-
-    setTimeout(abort, ABORT_DELAY);
-  });
-}
-
-function handleBrowserRequest(
-  request: Request,
-  responseStatusCode: number,
-  responseHeaders: Headers,
-  remixContext: EntryContext
-) {
-  return new Promise((resolve, reject) => {
-    let shellRendered = false;
-    const { pipe, abort } = renderToPipeableStream(
-      <RemixServer
-        context={remixContext}
-        url={request.url}
-        abortDelay={ABORT_DELAY}
-      />,
-      {
-        onShellReady() {
-          shellRendered = true;
-          const body = new PassThrough();
-          const stream = createReadableStreamFromReadable(body);
+        onAllReady() {
+          if (prohibitOutOfOrderStreaming) {
```

Comment:
mhmm - I'd personally vote for something like this to DRY it up a tiny bit if the two paths are identical (and have been for sometime?), but not a refactor I'd fight super hard for :)

```js
let readyOption =
  (userAgent && isbot(userAgent)) || remixContext.isSpaMode
    ? "onAllReady"
    : "onShellReady";

...

    const { pipe, abort } = renderToPipeableStream(
      <RemixServer
        context={remixContext}
        url={request.url}
        abortDelay={ABORT_DELAY}
      />,
      {
        [readyOption]() {
          shellRendered = true;
          ...
        }
```

Reply:
I like the way you cleaned this up. I've also refactored it so that the `readyOption` var is defined as close to its usage as possible so it's a bit clearer what's going on, and also added a type annotation to ensure it's a valid option name.